packages/sdk/src/dataProtector/sharing/addToCollection.ts

```diff
@@ -0,0 +1,166 @@
+import { getCollectionContract } from './smartContract/getCollectionContract.js';
+import { getProtectedDataById } from './subgraph/getProtectedDataById.js';
+
+export const addToCollection = async ({
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
+  sharingContractAddress,
+  protectedDataAddress: existingProtectedDataAddress,
+  collectionId,
+  onStatusUpdate,
+}: IExecConsumer &
+  SubgraphConsumer & {
+    dataProtectorContractAddress: AddressOrENS;
+    sharingContractAddress: AddressOrENS;
+  } & AddToCollectionParams): Promise<void> => {
+  // TODO: How to check that onStatusUpdate is a function?
+  // Example in zod: https://zod.dev/?id=functions
+  // const vonStatusUpdate: string = fnSchema().label('onStatusUpdate').validateSync(onStatusUpdate);
+
+  const vProtectedDataAddress = addressSchema()
+    .required()
+    .label('protectedDataAddress')
+    .validateSync(existingProtectedDataAddress);
+  const vCollectionId = positiveNumberSchema()
+    .required()
+    .label('collectionId')
+    .validateSync(collectionId);
+
+  const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+
+  const protectedData = await checkAndGetProtectedData({
+    graphQLClient,
+    protectedDataAddress: vProtectedDataAddress,
+    userAddress,
+  });
+
+  await checkCollection({
+    iexec,
+    sharingContractAddress,
+    collectionId: vCollectionId,
+    userAddress,
+  });
+
+  onStatusUpdate?.({
+    title: 'Give ownership to the collection smart-contract',
+    isDone: false,
+  });
+  // Approve collection SC to change the owner of my protected data in the registry SC
+  await approveCollectionContract({
+    iexec,
+    protectedData,
+    sharingContractAddress,
+  });
+  onStatusUpdate?.({
+    title: 'Give ownership to the collection smart-contract',
+    isDone: true,
+  });
+
+  onStatusUpdate?.({
+    title: 'Add protected data to your collection',
+    isDone: false,
+  });
+  await addProtectedDataToCollection({
+    iexec,
+    sharingContractAddress,
+    collectionId: vCollectionId,
+    protectedDataAddress: vProtectedDataAddress,
+  });
+  onStatusUpdate?.({
+    title: 'Add protected data to your collection',
+    isDone: true,
+  });
+};
+
+async function checkAndGetProtectedData({
+  graphQLClient,
+  protectedDataAddress,
+  userAddress,
+}: {
+  graphQLClient: GraphQLClient;
+  protectedDataAddress: Address;
+  userAddress: Address;
+}) {
+  const protectedData = await getProtectedDataById({
+    graphQLClient,
+    protectedDataAddress,
+  });
+
+  if (!protectedData) {
+    throw new ErrorWithData(
+      'This protected data does not exist in the subgraph.',
+      { protectedDataAddress }
+    );
+  }
+
+  if (protectedData.owner.id !== userAddress) {
+    throw new ErrorWithData('This protected data is not owned by the user.', {
+      protectedDataAddress,
+      userAddress,
+    });
+  }
+
+  return protectedData;
```

Comment:
i think we didn't need to return anything here

Reply:
Yes that seems strange but that makes it convenient for the `approveCollectionContract` function (called later) to check the collection's owner.
I'll change the code to make it a bit more explicit.